typings/component.type.ts

```diff
@@ -214,3 +214,41 @@ ComponentNext.lift({color: 'red'})
 
     return true
   })
+
+// $ExpectType ListComponentState<{ a: string; }>
+$(
+  ComponentNext.lift({a: ''})
+    .render((_, p: {propVal: string}) => p.propVal)
+    .toList(props => {
+      // $ExpectType { propVal: string; }
+      props
+
+      return props.propVal
+    })
+).iState
+
+// $ExpectType ListComponentState<{ b: string; a: string; }>
+$(
+  ComponentNext.lift({a: ''})
+    .matchR('action1', (value, state) => ({...state, b: ''}))
+    .render((_, p: {propVal: string}) => p.propVal)
+    .toList(props => props.propVal)
+).oState
+
+// $ExpectType Action<Action<unknown, "action1"> | Action<string, "action2">, string>
+$(
+  ComponentNext.lift({a: ''})
+    .matchR('action1', (value, state) => ({...state, b: ''}))
+    .matchC('action2', (value: string, state) => action('output', value))
+    .render((_, p: {propVal: string}) => p.propVal)
```

Comment:
we don't need `render` or the first `matchR` in this tests. I think they are covered in the other tests above.